### PR TITLE
Add timestamp checker

### PR DESCRIPTION
**Pull request by harryz2000**

### CODE REVIEW

File: include/TimeStampChecker.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#ifndef TIMESTAMPCHECKER_H
+#define TIMESTAMPCHECKER_H
+
+#include <media.h>
+#include <acumulator.h>
+
+class TimeStampChecker
+{
+public:
+	static constexpr uint32_t MaxDurationDiffMs = 3000;
```
Comment (murillo128) [inline include/TimeStampChecker.h:10]: this should be a configurable parameter on the constructor of `TimeStampChecker`

Comment (murillo128) [reply include/TimeStampChecker.h:10]: Also, please add a brief description of what does this class do

**Review by murillo128 — CHANGES_REQUESTED** [review]

File: include/TimeStampChecker.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	static constexpr uint32_t MaxContinousInvalidFrames = 5;
+	
+	enum class CheckResult
+	{
+		Valid,
+		Invalid,
+		Reset
+	};
+	
+	CheckResult Check(const MediaFrame &frame)
```
Comment (murillo128) [inline include/TimeStampChecker.h:20]: If it is a timestamp checked, pass the timestamps as input, not the whole frame

**Review by murillo128 — CHANGES_REQUESTED** [review]

File: include/TimeStampChecker.h [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	{
+		// Check if it is just started
+		if (lastRecvTime == 0 && lastTimeStamp == 0)
+		{
+			lastTimeStamp = timestamp;
+			lastRecvTime = recvTimeMs;
+			return CheckResult::Valid;
+		}
+		
+		auto durationOnTs = (timestamp - lastTimeStamp) * 1000 / clock;
```
Comment (murillo128) [inline include/TimeStampChecker.h:51]: either auto is an int64, and then we have propblems if timestamp - lastTimestamp value is too big, or it is and uint64_t and we have problems when timestamp < lastTimestamp.

Comment (harryz2000) [reply include/TimeStampChecker.h:51]: Yes. Missed this one. Will change to get the diff for whichever is larger.

Comment (harryz2000) [reply include/TimeStampChecker.h]: Changed to use a function to always get positive diff.

**Comment by harryz2000** [comment]
The issue wasn't reproduced in new dev server versions. Close for now.

**Review by murillo128 — APPROVED** [review]
lgtm

**Pull request merged**